common/metrics/src/metric_server.rs

```diff
@@ -45,11 +45,11 @@ fn whitelist_metrics(
 // only return the whitelisted metrics
 fn whitelist_json_metrics(
     json_metrics: HashMap<String, String>,
-    whitelist: Vec<String>,
+    whitelist: &'static [&'static str],
 ) -> HashMap<String, String> {
     let mut whitelist_metrics: HashMap<String, String> = HashMap::new();
     for (key, val) in json_metrics.iter() {
-        if whitelist.contains(&key) {
+        if whitelist.contains(&&key[..]) {
             whitelist_metrics.insert(key.to_string(), val.to_string());
         }
     }
```

Comment:
It might be more performant to iterate on the whitelist and lookup the key in the json_metrics hashmap since you'll only have to iterate through 2 items instead of however many are in that hashmap. might also be able to eliminate the funny `&&[..]` thing.